lib/driver.js

```diff
@@ -283,10 +283,19 @@ class AndroidUiautomator2Driver extends BaseDriver {
       this.caps.appPackage = appInfo.appPackage;
     }
 
+    try {
+      const pids = (await this.adb.getPIDsByName('uiautomator')).map((p) => `${p}`);
+      if (!_.isEmpty(pids)) {
+        await this.adb.shell(['kill', '-9', ...pids]);
```

Comment:
I thought device should be rooted to execute kill command. Am I wrong?

Reply:
I asume it's only necessary if the process to which the signal is sent is running under `root` account, but uiautomator is usually running under `shell` account, so it shouldn't be the case.

Reply:
Emu API 27
```
x:\>adb shell "ps | grep top" && adb shell kill -9 5884
shell         6528  5884    7832   3324 poll_schedule_timeout eb4fcaf0 S top
/system/bin/sh: kill: 5884: Permission denied
x:\>adb shell "ps | grep uiauto" && adb shell kill -9 7121
u10_a125      7121  1463 1037536  31372 ep_poll             0 S io.appium.uiautomator2.server
/system/bin/sh: kill: 7121: Operation not permitted
```